### ISSUE TITLE
The order of the -ldl option is out of place at run-time

### ISSUE THREAD

**Issue by aeurielesn** [coord-e/magicpak#2]
I noticed that the command being executed has the `-ldl` out of place w.r.t. the [source code](https://github.com/coord-e/magicpak/blob/3501b221080a82aae18723b0e36f1d9121058b29/src/domain/executable/resolver.rs#L164).

    [DEBUG] command: "/usr/bin/cc" "-Wl,-dynamic-linker,/lib64/ld-linux-x86-64.so.2" "-ldl" "-o" "/tmp/magicpak_resolver_14160520991579680778_3249826909874418912" "-xc" "/tmp/.tmpsNXfuc"

However, the following command succeeds:

    $ "/usr/bin/cc" "-Wl,-dynamic-linker,/lib64/ld-linux-x86-64.so.2" "-o" "/tmp/magicpak_resolver_14160520991579680778_3249826909874418912" "-xc" "/tmp/.tmpsNXfuc" "-ldl"
    $ echo $?
    0

My current workaround is to create a wrapper script around GCC with a hardcoded `-ldl` as the last option and pass it to `magicpack` using `--cc`.

**Comment by y0grt** [comment]
Could you explain your solution please?

I think is related to the problem I have, no matters what binary I try to pack:

```
[INFO] exe: loading /bin/bash
[INFO] action: bundle shared object dependencies of /bin/bash
[ERROR] error: Error happend during the compilation of library resolver: /usr/bin/ld: /tmp/ccJwTMhL.o: in function `main':
.tmpRlHKeg:(.text+0x3c): undefined reference to `dlopen'
/usr/bin/ld: .tmpRlHKeg:(.text+0x53): undefined reference to `dlerror'
/usr/bin/ld: .tmpRlHKeg:(.text+0x7a): undefined reference to `dlinfo'
/usr/bin/ld: .tmpRlHKeg:(.text+0x8a): undefined reference to `dlerror'
/usr/bin/ld: .tmpRlHKeg:(.text+0xb8): undefined reference to `dlclose'
collect2: error: ld returned 1 exit status
```

**Comment by aeurielesn** [comment]
@jmlgo
I'm using the following wrapper script:

    $ cat gcc-wrapper 
    #!/usr/bin/env bash
    $CC $@ -ldl

I'm using it this way:

    $ CC=gcc-9 ./magicpak --cc ./gcc-wrapper target bundle/

**Comment by aeurielesn** [comment]
This seems to be fixed on the develop branch but it hasn't been released.

**Comment by coord-e** [comment]
Sorry for the late reply. I've just released a new version with corresponding fix (#1) so the fixed binary will be available soon.

**Comment by aeurielesn** [comment]
Thanks
